### ISSUE TITLE
Why the model is different between loaded from pth and created from yaml?

### ISSUE THREAD

**Issue by yt7589** [ultralytics/ultralytics#4290]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I found that if I built a model by loading from the official weights yolov5s.pt the parameters of the model as below:
```bash
model.0.conv.weight: <class 'torch.nn.parameter.Parameter'>;
model.0.conv.bias: <class 'torch.nn.parameter.Parameter'>;
model.1.conv.weight: <class 'torch.nn.parameter.Parameter'>;
model.1.conv.bias: <class 'torch.nn.parameter.Parameter'>;
......
```
It seemed that the model had bias is True and without batch norm layer.

If I built a model by yolov8.yaml the parameters of the model as below:
```bash
model.0.conv.weight: <class 'torch.nn.parameter.Parameter'>
model.0.bn.weight: <class 'torch.nn.parameter.Parameter'>
model.0.bn.bias: <class 'torch.nn.parameter.Parameter'>
model.1.conv.weight: <class 'torch.nn.parameter.Parameter'>
model.1.bn.weight: <class 'torch.nn.parameter.Parameter'>
model.1.bn.bias: <class 'torch.nn.parameter.Parameter'>
```
It seemed that the model had bias set to False and with batch norm layer. 

I had checked the source code:
```python
class Conv(nn.Module):
    """Standard convolution with args(ch_in, ch_out, kernel, stride, padding, groups, dilation, activation)."""
    default_act = nn.SiLU()  # default activation

    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, d=1, act=True):
        """Initialize Conv layer with given arguments including activation."""
        super().__init__()
        self.conv = nn.Conv2d(c1, c2, k, s, autopad(k, p, d), groups=g, dilation=d, bias=False)
        self.bn = nn.BatchNorm2d(c2)
        self.act = self.default_act if act is True else act if isinstance(act, nn.Module) else nn.Identity()

    def forward(self, x):
        """Apply convolution, batch normalization and activation to input tensor."""
        return self.act(self.bn(self.conv(x)))

    def forward_fuse(self, x):
        """Perform transposed convolution of 2D data."""
        return self.act(self.conv(x))
```
It meant that the model should have bias set to False and have batch norm layer.

My question is why the model from pth and the model from yaml is different?

### Additional

If I did not modified the model structure and load the weights from official pth file it would generate different results from the model load from the pth directly. If I model the source code of Conv class to set bias to True and comment out self.bn relate lines it would generate the same results with the model load directly from pth file.

**Comment by glenn-jocher** [comment]
@yt7589 thank you for your detailed post. It seems like what you're observing is the difference in architecture between models trained with different settings. When you build a model from the pth file, it comes with the architecture that was used during the training of that model, which might include bias parameters and no BatchNorm layers depending on the nature and configuration of the training.

On the other hand, when you build a model from the yaml file, you're essentially building a fresh model using the architecture defined in the yaml file. The architecture in the yaml file and the training settings may include or exclude certain parameters or layers (like bias parameters or BatchNorm layers) depending on the specific configurations.

This discrepancy can potentially lead to different inference results even when using the same weights, due to the difference in model architecture and the presence/absence of certain layers or parameters.

If you modify the Conv class to match the architecture of the one used in the model built from the pth file (by setting bias to True and commenting out BatchNorm related lines), these models would indeed generate consistent results, since they now share the same architecture.

Please note that any change in the model architecture may impact the performance of the model, so it's important to understand the implications before making such changes.
